app/src/main/java/com/infomaniak/swisstransfer/ui/utils/Extensions.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Infomaniak SwissTransfer - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.swisstransfer.ui.utils
+
+import android.content.Context
+import android.content.Intent
+import androidx.compose.ui.platform.ClipboardManager
+import androidx.compose.ui.text.AnnotatedString
+
+object Extensions {
+
+    fun ClipboardManager.copyText(text: String) {
+        setText(AnnotatedString(text))
+    }
```

Comment:
You can do something like this to avoid having to get the ClipboardManager inside the compose code. Also this will prevent developers from forgetting to send a feedback to the user after a successful copy

```suggestion
    fun Context.copyText(text: String, showSnackbar: (String) -> Unit) {
        val clipboardManager = getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
        clipboardManager.setPrimaryClip(ClipData.newPlainText(text, text))

        showSnackbar(getString(R.string.textCopiedTODO))
    }
```